functions/src/lobby.ts

```diff
@@ -159,16 +161,31 @@ export const joinLobby = functions.https.onCall((data: unknown, context): Promis
       userInfo.avatar = AVATARS[Math.floor(Math.random() * AVATARS.length)];
     }
 
-    // add player
-    transaction.update(lobby, {
-      players: firestore.FieldValue.arrayUnion({
-        displayName: userInfo.displayName,
-        avatar: userInfo.avatar,
-        alive: true,
-        votes: 0,
-      }),
-      uids: firestore.FieldValue.arrayUnion(auth.uid),
-    });
+    // add spectator
+    if (lobbyData?.state != "WAIT") {
```

Comment:
check if lobbyData is undefined and throw an error so you don't have to do lobbyData?

Reply:
moved lobbyData to where I add spectator on line 164, if lobbyInfo is undefined it will throw an error, is that good?